app/Models/User.php

```diff
@@ -909,14 +924,29 @@ public function defaultGroup()
     public function groupIds()
     {
         return $this->memoize(__FUNCTION__, function () {
-            return $this->userGroups->pluck('group_id')->toArray();
+            $ret = [
+                'active' => [],
+                'pending' => [],
+            ];
+
+            foreach ($this->userGroups as $userGroup) {
+                $key = $userGroup->user_pending ? 'pending' : 'active';
+                $ret[$key][] = $userGroup->group_id;
+            }
+
+            return $ret;
         });
     }
 
     // check if a user is in a specific group, by ID
-    public function isGroup($group)
+    public function isGroup($group, $checkToken = true)
```

Comment:
I'd name the param to something else that more accurately reflects whether the function is being used as a permission check (in no groups if using oauth), or as an existence check. 
`isGroupPending` probably doesn't need the same options since it wouldn't make sense to do a permission check on a pending group? :thinking:

Reply:
can use some function name suggestion :thinking:

Reply:
`$forPermission` or `checkPermissions` or something? Or just have a separate `hasGroupPermission` function instead 🤔